slowmovie.py

```diff
@@ -105,15 +105,44 @@ def video_info(file):
         frameTime = 1000 / fps
 
         info = {
-            'frame_count': frameCount,
-            'fps': fps,
-            'duration': duration,
-            'frame_time': frameTime}
+            "frame_count": frameCount,
+            "fps": fps,
+            "duration": duration,
+            "frame_time": frameTime}
 
         videoInfos[file] = info
     return info
 
 
+# Returns the next video in the videos directory, or the first one if there's no current video
+def get_next_video(viddir, currentVideo=None):
+    # Only consider videos in the directory
+    videos = sorted(list(filter(supported_filetype, os.listdir(viddir))))
+
+    # Return None if there are no videos
+    if not videos:
+        return None
+
+    if currentVideo:
+        try:
+            nextIndex = videos.index(currentVideo) + 1
+            # If we're not wrapping around
+            if not nextIndex >= len(videos):
+                return os.path.join(viddir, videos[nextIndex])
+        except ValueError:
+            # Maybe currentVideo is no longer in the list and that's fine
+            pass
```

Comment:
How would it end up here? If the current video is deleted/moved/renamed while it's running, wouldn't `generate_frame()` fail?

Reply:
I wonder if this was more likely in previous versions when the video directory was read at program start and then never refreshed? The chance of the video being deleted is not 0 but it's extremely small. You'd have to delete it between the generate frame call but prior to this, that window is probably measured in milliseconds.

Reply:
Yeah, I think that's what happened here. I'll update the code. Thanks for catching this :)